### ISSUE TITLE
wangqr/Aegisub as the new Debian upstream

### ISSUE THREAD

**Issue by aniolm9** [wangqr/Aegisub#13]
Hello,
I couldn't find any other way to contact you, so I'm opening this issue.

I'm the Aegisub Debian Maintainer. The official project has been inactive for a while and it makes the Debian package maintenance quite difficult. We have decided to use an active fork as the new upstream, and we are considering this one. What do you think about it?

Thanks,
Aniol

**Comment by wangqr** [comment]
Sure you can use this one as new upstream.

#### Few things about the CMake build:

In original repo, some dependencies are included in the `vendor` folder. On linux it should be possible to use system libs instead of building from `vendor`. But I'm using Windows mostly, and this makes it hard to upgrade dependencies. So I migrated to CMake. The original configure / makefile is not touched, but I haven't verified if they are still working. Hopefully they will work as before.

If you are interested in using CMake, the [travis script](https://github.com/wangqr/Aegisub/blob/dev/.travis.yml) and [Arch PKGBUILD](https://github.com/wangqr/PKGBUILDs/blob/master/aegisub-git/PKGBUILD) may be helpful. I plan to automate copying the `automation` folder and translations, but currently this NOT included in the CMake script and needs to be done manually.

#### Few things about me:

I am maintaining this fork because some of my friends are using it and found some issues. And this fork is maintained by myself in spare time. So I may maintain it less frequently if I'm not using it anymore or I get busy.

As for contacting me, you can open an issue, or my email address can be found in `git log`.

**Comment by aniolm9** [comment]
**About building:**

I've tried to build using the original configure/Makefile and it crashes, [see the log](https://gist.github.com/aniolm9/b293a9235d615a64746f2341c54db300). It may be be related with this [issue](https://github.com/Aegisub/Aegisub/issues/120). With CMake I can build, but as you say, on Linux system libs should be used. I still have to figure out if it is easier to patch what makes the Makefile crash or CMake to not use the `vendor` directory.

I have also noticed that translations aren't included, but I'm happy to read that you are planning to work on it.

**About the repo:**

Besides the usual maintenance, to use it as an upstream for Debian, releases are required. Would you be able to do that?

On the other hand, are you the only one who can merge, release, etc.?

**Comment by wangqr** [comment]
> It may be be related with this issue.

It is not related to the main issue (OpenGL on HiDPI), but it is mentioned in comment. Looks like somehow `configure` fails to find some libs (errors are in `audio_player.cpp`, so it fails to find any of `alsa`, `openal`, `oss`, `portaudio`, or `libpulse`). See [1](https://github.com/Aegisub/Aegisub/issues/120#issuecomment-493630689) [2](https://github.com/Aegisub/Aegisub/issues/120#issuecomment-493631576).

> With CMake I can build, but as you say, on Linux system libs should be used.

CMake already use system libs, except `luajit` and `csri` are still built from `vendor`. The bundled version of these 2 dependencies are also used when building using configure/makefile.
* `libaegisub` is statically linked to `luajit` and needs some work if we want to migrate to official release
* `csri` is a standard interface to communicate with subtitle renderers. It is not maintained as separate project elsewhere. MPC-HC is another project using csri, and it also embeds csri as [a part of the project](https://github.com/mpc-hc/mpc-hc/blob/master/src/filters/transform/VSFilter/csri.h).

> Besides the usual maintenance, to use it as an upstream for Debian, releases are required. Would you be able to do that?

Yes, I can do that. I'll evaluate the workload to decide whether I can include those missing parts to CMake before release.

> On the other hand, are you the only one who can merge, release, etc.?

Yes.

**Comment by wangqr** [comment]
Just fixed the Makefile issue. Verified on my Archlinux machine.

**Comment by aniolm9** [comment]
> CMake already use system libs, except `luajit` and `csri` are still built from `vendor`. The bundled version of these 2 dependencies are also used when building using configure/makefile.
> 
I have managed to build with CMake deleting all the directories under `vendor` except from `luabins`, `luajit` and `csri`. I have also managed to build with configure/makefile keeping just `luabins` and `luajit`.
> 
> Yes, I can do that. I'll evaluate the workload to decide whether I can include those missing parts to CMake before release.
> 
I have very few experience with CMake and C++, but I can help with releases and some other maintenance stuff if needed.

**Comment by wangqr** [comment]
Since Makefile works fine now, I'll just do the release after some final check. CMake will be left there without automatically copying automation and translation (they are automated when using Makefile). Does this plan looks fine?

> I have also managed to build with configure/makefile keeping just luabins and luajit.

`csri` is an optional function and can be turned off. But it is just a few headers that defines the function prototype and never changed after initial release. Any dynamic lib that exports those functions are csri-compatible. It is not separately maintained. So people just include those header within their project. If you remove them, then Aegisub will not support csri.

PS: Currently some files in `vendor/csri` are from from `asa` which is a dead subtitle renderer that supports csri. Aegisub is only using the csri header, so `asa` can be removed but haven't been done yet.
PS2: You can find a list of subtitle renderers [here](https://github.com/libass/libass#other-assssa-implementations). `csri` is mainly for people who want to use VSFilter instead of the default libass. But VSFilter itself is Windows only. Actually I haven't seen anyone using csri-compatible shared object on linux, except VSFilter over [wine](https://www.winehq.org/). So if you decided to remove `csri` from the linux build, for most users it won't break anything.

**Comment by aniolm9** [comment]
> Since Makefile works fine now, I'll just do the release after some final check. CMake will be left there without automatically copying automation and translation (they are automated when using Makefile). Does this plan looks fine?
> 
Yes. This weekend I will package a new stable version to fix the SEGV with the spell checking and then I will start working with the upstream replacement.
> 
> `csri` is an optional function and can be turned off. But it is just a few headers that defines the function prototype and never changed after initial release. Any dynamic lib that exports those functions are csri-compatible. It is not separately maintained. So people just include those header within their project. If you remove them, then Aegisub will not support csri.
> 
Thanks for the explanation. I've checked and the versions 3.2.* in Debian don't include `csri`, so I don't think I'm going to include it.

I will close this issue after the first release to Debian using this repository.

**Comment by wangqr** [comment]
As you may know, @tgoyne has just merged lots of commits to Aegisub/Aegisub, including those critical ones which are the reason of the existence of this fork. If the upstream is actively maintained again, I'll just leave this fork here and make PRs to the upstream.

I'll continue maintaining this fork at least until a new binary windows build is made by the upstream.

**Comment by aniolm9** [comment]
Yes, I saw that. I will think about what to do next.

Thanks!

**Comment by aniolm9** [comment]
Hi,
It seems that Aegisub upstream is almost dead. Even the official website (aegisub.org) has been down for a while. @wangqr if you still want to maintain this fork I'll change Debian upstream next week.

Regards!

**Comment by wangqr** [comment]
You might want to contact @CoffeeFlux . They have write access to the upstream repository, and emailed me a few months ago regarding releasing 3.4.0 from upstream after merging bug fixes. Issue Aegisub/Aegisub#214 was then opened to track the progress.

**Comment by Cookiesss555** [comment]
Is there any update on this? Using this fork as the new upstream.
I'm trying to install aegisub on Ubuntu 22.04 LTS and it says no package found since the [build failed to build from source](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=997098). It would be nice to be able to install it easily like in 20.04.
